rkt/pubkey/pubkey.go

```diff
@@ -227,6 +235,36 @@ func downloadKey(u *url.URL) (*os.File, error) {
 	return retTf, nil
 }
 
+func getClient(skipTLSCheck bool) *http.Client {
+	if !skipTLSCheck {
+		return http.DefaultClient
+	}
+	client := *http.DefaultClient
+	var tr *http.Transport
+	// default transport is hidden behind the RoundTripper
+	// interface, so we can't easily make a copy of it
+	if transport, ok := http.DefaultTransport.(*http.Transport); ok {
```

Comment:
What's the point of these machinations? Why do we need divergent behaviour based on go version?